token-erc-20/chaincode-java/src/main/java/org/hyperledger/fabric/samples/erc20/ERC20TokenContract.java

```diff
@@ -0,0 +1,444 @@
+import org.hyperledger.fabric.shim.ledger.CompositeKey;
+
+@Contract(
+    name = "erc20token",
+    info =
+        @Info(
+            title = "ERC20Token Contract",
+            description = "The erc20 fungible token implementation.",
+            version = "0.0.1-SNAPSHOT",
+            license =
+                @License(
+                    name = "Apache 2.0 License",
+                    url = "http://www.apache.org/licenses/LICENSE-2.0.html"),
+            contact =
+                @Contact(
+                    email = "renjithkn@gmail.com",
+                    name = "Renjith Narayanan",
+                    url = "https://hyperledger.example.com")))
+@Default
+public final class ERC20TokenContract implements ContractInterface {
+
+  /**
+   * Mint creates new tokens and adds them to minter's account balance. This function triggers a
+   * Transfer event.
+   *
+   * @param ctx the transaction context
+   * @param amount of tokens to be minted
+   */
+  @Transaction(intent = Transaction.TYPE.SUBMIT)
+  public void Mint(final Context ctx, final long amount) {
+
+    // Check minter authorization - this sample assumes Org1 is the central banker with privilege to
+    // mint new tokens
+    String clientMSPID = ctx.getClientIdentity().getMSPID();
+    ChaincodeStub stub = ctx.getStub();
+    if (!clientMSPID.equalsIgnoreCase(ContractConstants.MINTER_ORG_MSPID.getValue())) {
+
+      throw new ChaincodeException(
+          "Client is not authorized to mint new tokens", UNAUTHERIZED_SENDER.toString());
+    }
+
+    // Get ID of submitting client identity
+    String minter = ctx.getClientIdentity().getId();
+    if (amount <= 0) {
+      throw new ChaincodeException(
+          "Mint amount must be a positive integer", INVALID_AMOUNT.toString());
+    }
+
+    CompositeKey balanceKey = stub.createCompositeKey(BALANCE_PREFIX.getValue(), minter);
+    String currentBalanceStr = stub.getStringState(balanceKey.toString());
+    // If minter current balance doesn't yet exist, we'll create it with a current balance of 0
+    long currentBalance = 0;
+    if (!stringIsNullOrEmpty(currentBalanceStr)) {
+      currentBalance = Long.parseLong(currentBalanceStr);
+    }
+    // Used safe math .
+    long updatedBalance = Math.addExact(currentBalance, amount);
+
+    stub.putStringState(balanceKey.toString(), String.valueOf(updatedBalance));
+    // Increase totalSupply
+    String totalSupplyStr = stub.getStringState(TOTAL_SUPPLY_KEY.getValue());
+    long totalSupply = 0;
+    if (!stringIsNullOrEmpty(totalSupplyStr)) {
+      totalSupply = Long.parseLong(totalSupplyStr);
+    }
+    // Used safe math .
+    totalSupply = Math.addExact(totalSupply, amount);
+    stub.putStringState(TOTAL_SUPPLY_KEY.getValue(), String.valueOf(totalSupply));
+    Transfer transferEvent = new Transfer("0x0", minter, amount);
+    stub.setEvent(TRANSFER_EVENT.getValue(), transferEvent.toJSONString().getBytes(UTF_8));
+  }

```

Comment:
Please could you add a log statement here (stdout, Logger, etc...) to indicate that the contract routine has successfully returned?  E.g., golang: 

```
	log.Printf("minter account %s balance updated from %d to %d", minter, currentBalance, updatedBalance)

	return nil
}
```

Each of the top-level / high-level invocations of the chaincode should print a small output to the log.  This is very helpful for debugging and troubleshooting the installation of CCaaS.

Reply:
Done